commit 178f99c8e899108e147fd741fad8b69da051465d
Author: popcornmix <popcornmix@gmail.com>
Date:   Mon Apr 29 19:16:16 2019 +0100

    Revert "staging: bcm2835-audio: use module_platform_driver() macro"
    
    commit ed4c2e5dc4216d5dded502bfcf594d3984e6bccd upstream.
    
    This reverts commit 786ced30fec053b27248ed5b24dcde61ed3f47f6.

diff --git a/drivers/staging/vc04_services/bcm2835-audio/bcm2835.c b/drivers/staging/vc04_services/bcm2835-audio/bcm2835.c
index b8e148ade6f6..da0fa34501fa 100644
--- a/drivers/staging/vc04_services/bcm2835-audio/bcm2835.c
+++ b/drivers/staging/vc04_services/bcm2835-audio/bcm2835.c
@@ -470,7 +470,25 @@ static struct platform_driver bcm2835_alsa0_driver = {
 		.of_match_table = snd_bcm2835_of_match_table,
 	},
 };
-module_platform_driver(bcm2835_alsa0_driver);
+
+static int bcm2835_alsa_device_init(void)
+{
+	int retval;
+
+	retval = platform_driver_register(&bcm2835_alsa0_driver);
+	if (retval)
+		pr_err("Error registering bcm2835_audio driver %d .\n", retval);
+
+	return retval;
+}
+
+static void bcm2835_alsa_device_exit(void)
+{
+	platform_driver_unregister(&bcm2835_alsa0_driver);
+}
+
+late_initcall(bcm2835_alsa_device_init);
+module_exit(bcm2835_alsa_device_exit);
 
 MODULE_AUTHOR("Dom Cobley");
 MODULE_DESCRIPTION("Alsa driver for BCM2835 chip");
diff -uN a/1.txt b/1.txt
--- a/dummy/rpi_1440_178f99c8e899108e147fd741fad8b69da051465d.txt	1970-01-01 00:00:00.000000000 +0000
+++ b/dummy/rpi_1440_178f99c8e899108e147fd741fad8b69da051465d.txt	2013-12-23 04:07:40.000000000 +0000
@@ -0,0 +1 @@
+dummy file to ensure patch has content.
